x/emissions/keeper/keeper.go

```diff
@@ -1160,6 +1172,16 @@ func (k *Keeper) AppendInference(
 		if err != nil {
 			return errorsmod.Wrap(err, "error calculating and saving inferer score ema with last saved topic quantile")
 		}
+
+		// Check if the inferer with lowest score is active before removing it, because remove will not fail if the inferer is not active
+		isActive, err := k.IsActiveInferer(ctx, topic.Id, lowestEmaScore.Address)
+		if err != nil {
+			return errorsmod.Wrap(err, "error checking if inferer is active")
+		}
+		if !isActive {
+			return errors.New("inferer with lowest score is not active")
+		}
+
```

Comment:
I'm wondering if it should be handled this way. If we are trying to remove an active inferer that doesn't exist, it's a bug on our end. Maybe it's just a sanity check to avoid disrupting the reward flow, but if this happens, we should be made aware somehow. Lmk your thoughts.

Reply:
This is an extra check indeed. "Defense in depth" applies here imo. Could def argue it's excessive